c_glib/arrow-dataset-glib/scanner.cpp

```diff
@@ -132,95 +113,59 @@ gadataset_scanner_to_table(GADatasetScanner *scanner,
-                                       const GValue *value,
-                                       GParamSpec *pspec)
-{
+static void gadataset_scanner_builder_set_property(GObject* object, guint prop_id,
+                                                   const GValue* value,
+                                                   GParamSpec* pspec) {
   auto priv = GADATASET_SCANNER_BUILDER_GET_PRIVATE(object);
 
   switch (prop_id) {
-  case PROP_SCANNER_BUILDER:
-    priv->scanner_builder =
-      *static_cast<std::shared_ptr<arrow::dataset::ScannerBuilder> *>(
-        g_value_get_pointer(value));
-    break;
-  case PROP_USE_ASYNC:
-    garrow::check(nullptr,
-                  priv->scanner_builder->UseAsync(g_value_get_boolean(value)),
-                  "[scanner-builder][use-async][set]");
-    break;
-  default:
-    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
-    break;
+    case PROP_SCANNER_BUILDER:
+      priv->scanner_builder =
+          *static_cast<std::shared_ptr<arrow::dataset::ScannerBuilder>*>(
+              g_value_get_pointer(value));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+      break;
   }
 }
 
-static void
-gadataset_scanner_builder_init(GADatasetScannerBuilder *object)
-{
+static void gadataset_scanner_builder_init(GADatasetScannerBuilder* object) {
   auto priv = GADATASET_SCANNER_BUILDER_GET_PRIVATE(object);
-  new(&priv->scanner_builder) std::shared_ptr<arrow::dataset::ScannerBuilder>;
+  new (&priv->scanner_builder) std::shared_ptr<arrow::dataset::ScannerBuilder>;
 }
 
-static void
-gadataset_scanner_builder_class_init(GADatasetScannerBuilderClass *klass)
-{
+static void gadataset_scanner_builder_class_init(GADatasetScannerBuilderClass* klass) {
   auto gobject_class = G_OBJECT_CLASS(klass);
-  gobject_class->finalize     = gadataset_scanner_builder_finalize;
+  gobject_class->finalize = gadataset_scanner_builder_finalize;
   gobject_class->set_property = gadataset_scanner_builder_set_property;
 
-  GParamSpec *spec;
-  spec = g_param_spec_pointer("scanner-builder",
-                              "Scanner builder",
-                              "The raw "
-                              "std::shared<arrow::dataset::ScannerBuilder> *",
-                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
-                                                       G_PARAM_CONSTRUCT_ONLY));
+  GParamSpec* spec;
+  spec = g_param_spec_pointer(
+      "scanner-builder", "Scanner builder",
+      "The raw "
+      "std::shared<arrow::dataset::ScannerBuilder> *",
+      static_cast<GParamFlags>(G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY));
   g_object_class_install_property(gobject_class, PROP_SCANNER_BUILDER, spec);
-
-  arrow::dataset::ScanOptions default_options;
-  /**
-   * GADatasetScannerBuilder:use-async:
-   *
-   * Whether or not async mode is used.
-   *
-   * Since: 6.0.0
-   */
-  spec = g_param_spec_boolean("use-async",
```

Comment:
Similarly, is there a way to deprecate things here instead of removing it?

Reply:
We can use `g_warning()` for deprecating.
But we can remove it without deprecating in this case because it was only used internally for now.